apps/somnia/src/commands/archive.ts

```diff
@@ -0,0 +1,263 @@
+      )
+  )
+
+//requiring all the data because imports dont work
+const archivepath = path.join(cwd, '/libs/gi/dm-localization/assets/locales/en')
+//get keys
+const archive: Record<string, any> = {
+  key: {
+    char: require(path.join(archivepath, '/charNames_gen.json')),
+    weapon: require(path.join(archivepath, '/weaponNames_gen.json')),
+    artifact: require(path.join(archivepath, '/artifactNames_gen.json')),
+  },
+  char: {},
+  weapon: {},
+  artifact: {},
+}
+//traveler data
+for (const name in archive['key']['char']) {
+  if (name.match(/Traveler/)) delete archive['key']['char'][name]
+}
+archive['key']['char']['TravelerAnemoF'] = 'Traveler (Anemo)'
+archive['key']['char']['TravelerGeoF'] = 'Traveler (Geo)'
+archive['key']['char']['TravelerElectroF'] = 'Traveler (Electro)'
+archive['key']['char']['TravelerDendroF'] = 'Traveler (Dendro)'
+archive['key']['char']['TravelerHydroF'] = 'Traveler (Hydro)'
+//get all the data from keys
+for (const category in archive['key']) {
+  for (const name in archive['key'][category]) {
+    const itempath = path.join(archivepath, `/${category}_${name}_gen.json`)
+    if (fs.existsSync(itempath)) archive[category][name] = require(itempath)
+  }
+}
+
+const colors = {
+  rarity: [0x818486, 0x5a977a, 0x5987ad, 0x9470bb, 0xc87c24],
+  element: {
+    none: {
+      img: 'https://api.ambr.top/assets/UI/UI_Icon_Item_Temp.png',
+      color: 0xaaaaaa,
+    },
+    anemo: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Wind.png',
+      color: 0x61dbbb,
+    },
+    geo: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Rock.png',
+      color: 0xf8ba4e,
+    },
+    electro: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Electric.png',
+      color: 0xb25dcd,
+    },
+    hydro: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Water.png',
+      color: 0x5680ff,
+    },
+    pyro: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Fire.png',
+      color: 0xff3c32,
+    },
+    cryo: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Ice.png',
+      color: 0x77a2e6,
+    },
+    dendro: {
+      img: 'https://api.ambr.top/assets/UI/UI_Buff_Element_Grass.png',
+      color: 0xa5c83b,
+    },
+  },
+}
+const talentlist = {
+  p: { name: 'Character Profile', value: 'p' },
+  n: { name: 'Normal/Charged/Plunging Attack', value: 'n' },
+  e: { name: 'Elemental Skill', value: 'e' },
+  q: { name: 'Elemental Burst', value: 'q' },
+  a: { name: 'Ascension Passives', value: 'a' },
+  c: { name: 'Constellations', value: 'c' },
+}
+
+//clean tags from input
+//discord has no colored text, so just bold everything instead
+function clean(s: string) {
+  //keep italic tags
+  s = s.replaceAll(/(<\/?i>)+/g, '*')
+  //turn rest into bold
+  s = s.replaceAll(/(<\/?\w+>)+/g, '**')
+  //ignore <br/> tags
+  s = s.replaceAll(/<\w+\/>/g, '')
+  //remove extra whitespace
+  return s.trim()
+}
+
+export { archive, clean, colors, talentlist }
```

Comment:
Just export these when you are creating them (unless this does something different im not aware of)

Reply:
specifically archive i can't export as i create it, since it gets changed after